doc/extension_store/README.md

```diff
@@ -0,0 +1,76 @@
+
+## Parameters to Store
+
+The Extension management component on the cluster must store the following information for an installed VS:
+
+* For each configuration:
+    * ID and version
+        * Required to detect if an older version is already installed
+        * Required to detect if the user tries to install another VS using the same configuration
+    * Paths to installed libraries and configuration files on BucketFS
+        * File names might change in newer versions.
+        * Needed to delete files from BucketFS during update/delete
+    * Parameters with `scope=configuration`
+        * Required when creating a new VS using the same configuration.
+        * Cannot be changed during update. Maybe hide them?
+* For each Connection / Virtual Schema:
+    * ID and version of the configuration
+        * Required to check if another VS uses the same configuration
+    * Parameters with `scope=connection` that the user entered during the installation process, required for updating the values.
+    * **Don't store** values for fields with `type=password`.
+        * User has to enter a new password when updating.
+
+## Installation Process
+
+1. If no other VS uses this configuration or the configuration was not installed before:
+    1. Ask user to enter values for all parameters with `scope=configuration`
+    1. If `licenseAgreementRequired=true` (usually for JDBC drivers): show the license and ask user to agree
+    1. Download all artifacts in the `bucketFsUpload`
+    1. Upload the artifacts to BucketFS, using the specified path names
+    1. Run the `create.configuration` SQL statements
+1. Ask user to enter values for all parameters with `scope=connection`
+1. Run the `create.connection` SQL statements.
+
+## Update Process
+
+1. Check if there is a newer version of the configuration. If yes:
+    1. Ask user to confirm the license again (usually the JDBC driver license)
+    1. Download new JARs
+    1. Delete old JARs from BucketFS
+    1. Install new JARs to BucketFS
+    1. Run scripts from `update.configuration`
+1. Let user enter new values for all parameters with `scope=connection`
+1. Run `update.connection` SQL statements.
+
+## Delete Process
+
+1. Run `drop.connection` SQL statements.
+1. If no other VS uses this configuration:
+    1. Run `drop.configuration` SQL statements.
+    1. Delete JARs from BucketFS (using stored paths)
+
+## Open Points
+
+* Current JSON structure and property names is just a draft and open for discussion!
+* What happens if the Adapter Schema or other resources already exists?
+* Simplify process by auto-generating names, e.g. `connectionName`, `adapterScriptName`.
+* How to detect if another VS already has installed some JARs?
+* How to handle when parameters with `scope=configuration` are added in newer versions?
+    * Don't allow this? Prompt the user for new values?
+* Allow renaming the VIRTUAL SCHEMA by updating?
```

Comment:
We have, potentially, a similar problem with the UDFs for prediction. Currently, we create one UDF per model. In the worst case, they all need to change (there might be many). Do we need some kind of patch mechanism? The probelm, these might be complicated and probably need code execution.

Reply:
I have no idea how to solve this.

Reply:
I think, the idea to solving this was, to keep the old versions if the extension and add new ones. That is maybe a thought, we should take in consideration. The udfs are defined for a specific language and we add new languages for each version. Maybe we can then also have a script which checks if the language is still used somewhere and warn the user, during extension deletion that he still uses the extension.

Reply:
To get the deletion check working, we probably also need a list of things, which scripts were created by the installer to decide if there are still UDF which the user created.

Reply:
Added to open points